Improve input-splitting and controller disconnect errors

Himanshu-01 opened the pull request:
* add basic task progress dialog during input split process

Himanshu-01 committed:
Rewrite `user_interface_controller_update`
 - improve controller disconnection handling

Himanshu-01 committed:
add basic task progress to stop 4way from taking excessive inputs during transition

Berthalamew commented on xlive/Blam/Engine/interface/screens/screen_main_menu.cpp:
```diff
@@ -337,6 +376,26 @@ void c_main_menu_list::handle_item_quit(s_event_record** pevent)
+	params.m_context = 0;
+	params.user_flags = user_interface_controller_get_signed_in_controllers_mask(); //replacing 0xFF with active_controllers so that controller-removed-handler stops panicking here
+	params.m_channel_type = _user_interface_channel_type_dialog;
+	params.m_screen_state.field_0 = 0xFFFFFFFF;
```
set these to NONE

Berthalamew commented on xlive/Blam/Engine/interface/user_interface_globals.h:
```diff
@@ -1,11 +1,13 @@
 #pragma once
 #include "user_interface.h"
 
+enum e_scenario_type : short;
```
int16

Berthalamew commented on xlive/Blam/Engine/input/input_process.cpp:
```diff
@@ -0,0 +1,20 @@
+void input_split_task_progress_callback(c_screen_xbox_live_task_progress_dialog* dialog)
+{
+	if (input_windows_has_split_device_active())
+		dialog->set_display_text_raw(L"Splitting in progress, please wait...");
```
add array for these messages for each language?

Himanshu-01 committed:
Add requested changes

Berthalamew committed:
misc changes

- set parameter to const in c_screen_xbox_live_task_progress_dialog::set_display_text_raw
